src/wallet/rpcwallet.cpp

```diff
@@ -3536,7 +3536,9 @@ UniValue getaddressinfo(const JSONRPCRequest& request)
         if (!meta->hdKeypath.empty()) {
             ret.pushKV("hdkeypath", meta->hdKeypath);
             ret.pushKV("hdseedid", meta->hd_seed_id.GetHex());
-            ret.pushKV("hdmasterkeyid", meta->hd_seed_id.GetHex());
+            if (!meta->master_key_id.IsNull()) {
+                ret.pushKV("hdmasterkeyid", meta->master_key_id.GetHex());
```

Comment:
`master_key_id` will often be only filled with the fingerprint rather than the entire ID, perhaps make a note in the help text above or only output the first 4 bytes in that case rather than a lot of zeroes too?

An alternative is to just always store the fingerprint in metadata, never the whole key, because I don't think it's used anywhere else.

Reply:
I've changed this to use the fingerprint by adding `KeyOriginInfo` to `CKeyMetadata`.